docs/distribution/launcher-cli.md

```diff
@@ -0,0 +1,194 @@
+
+> The actionables for this document are:
+>
+> - This document is a draft. The explanations are just to give an idea of the
+>   commands. It will be updated with detailed explanations when the CLI is
+>   developed.
+
+<!-- MarkdownTOC levels="2,3" autolink="true" -->
+
+- [Commands](#commands)
+  - [`new`](#new)
+  - [`install`](#install)
+  - [`uninstall`](#uninstall)
+  - [`list`](#list)
+  - [`default`](#default)
+  - [`config`](#config)
+  - [`run`](#run)
+  - [`repl`](#repl)
+  - [`language-server`](#language-server)
+  - [`upgrade`](#upgrade)
+  - [`version`](#version)
+  - [`help`](#help)
+- [General Options](#general-options)
+  - [`--path`](#--path)
+  - [`--version`](#--version)
+  - [`--use-system-jvm`](#--use-system-jvm)
+
+<!-- /MarkdownTOC -->
+
+## Commands
+
+### `new`
+Create a new, empty project in a specified directory.
+By default uses the `default` enso version, which can be overriden with
+`--version`.
+
+Examples:
+```bash
+> enso new project1 --path /somewhere/on/the/filesystem
+    # creates project called project1 in the specified directory
+    # using the `default` Enso version
+> enso new project2 --version 2.0.1
+    # creates the project in the current directory, using the 2.0.1 version
+```
+
+### `install`
+Installs a specific version of Enso.
+
+Examples:
+```bash
+> enso install 2.0.1
+```
+
+### `uninstall`
+Uninstalls a specific version of Enso.
+
+Examples:
+```bash
+> enso uninstall 2.0.1
+```
+
+### `list`
+Lists all installed versions of Enso and managed GraalVM distributions.
+
+Optional arguments are `enso` to just list Enso installations or `runtime` to
+list the installed runtimes.
+
+Examples:
+```bash
+> enso list
+Enso 2.0.1 -> GraalVM 20.1.0-java11
+Enso 2.2.3 -> GraalVM 20.1.0-java11
+Enso 3.0.0 -> GraalVM 21.1.0-java14
+> enso list enso
+2.0.1
+2.2.3
+3.0.0
+> enso list runtime
+GraalVM 20.1.0-java11 (used by 2 Enso installations)
+GraalVM 21.1.0-java14 (used by 1 Enso installation)
+```
+
+### `default`
+Sets the default Enso version used outside of projects.
+
+If run without arguments, displays currently configured `default` version.
+
+Examples:
+```bash
+> enso default 2.0.1
+default set to version 2.0.1 
+> enso default
+default version is 2.0.1
+```
+
+### `config`
+Can be used to manage project configuration or global user configuration (if
+outside a project or with the `--global` flag).
+
+If only the config path is provided, currently configured value is printed.
+
+Examples:
+```bash
+> enso config --global user.name Example User
+> enso config author.name Example User
+> enso config author.name
+Example User
+``` 
+
+### `run`
+Runs a project or an Enso script file.
+
+Examples:
+```bash
+> enso run script.enso # runs the file in script mode
+> enso run path/to/project1 # runs the project
+> enso run # runs the current project based on current directory
+``` 
+
+### `repl`
+Launches an Enso repl.
+
+Examples:
+```bash
+> enso repl
```

Comment:
This should probably take the same arguments as `run`? So that we can drop a repl in a random project and play around with it.

Reply:
I thought that currently the runner does not support that.

But if such a thing is required it can be added to the runner, so I will update the doc accordingly.

Reply:
Let me dream. The runner does not support 80% of this document, it looks more like "what I want it to be". So I want it to work :P

Reply:
That was more or less the idea of the global `--path` parameter - that we can add it to REPL to run the repl inside a project.
Should it take an optional project path in the format `enso repl path/to/project` instead?

I understand that the REPL should also be launched 'inside' a project if `enso repl` is called with a working directory that is inside the directory structure of some project, is that correct?